client/components/domains/use-my-domain/utilities/get-domain-name-validation-error-message.js

```diff
@@ -1,11 +1,18 @@
 import { createElement, createInterpolateElement } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 
+const isIdnDomain = ( domainName ) =>
+	/^(?!-)(?:[\p{L}0-9-]{0,61}[\p{L}0-9]\.)+[\p{L}]{2,}$/u.test( domainName );
```

Comment:
Just to ensure I understand this regex 😬 `^(?!-)` means that the string can't start with a dash, is that right?

Reply:
Yup. Any string that doesn't start with whatever comes after `?!`